HIP-0006.md

```diff
@@ -0,0 +1,170 @@
+
+- When the user wants to bid, their wallet creates a "perfect" UTXO by spending
+to itself, creating a coin with the exact value desired. This output can be used
+(even before it is confirmed) in this protocol.
+
+### Auction FINALIZE
+
+The name owner collects partially-signed bids and after some time, declares the highest
+value bid the winner. The name owner completes the auction by adding a payment output
+to the partially-signed transaction, signing and broadcasting:
+
+```
+vin:
+  0: current name owner, encumbered by Swap Script
+  1: A UTXO controlled by the bidder of some value V
+vout:
+  0: TRANSFER to the bidder's address
+  1: payment to the name-owner
+```
+
+The name owner pays themselves from the fully funded transaction, and signs input 0
+with `SIGHASH_ALL`. This transaction is a valid atomic swap and can be broadcast to the network.
+Because the locking script prevents all other name actions, the auction winner can
+`FINALIZE` this name transfer any time after the 288-block lockout without any other
+obligation.
+
+Note that since the bidder creates the INPUT and the name owner creates the OUTPUT,
+the name owner MUST also pay the miner fee for this transaction. They do so by reducing
+the value of output 1, effectively paying the miner fee out of the money they earned
+by the sale.
+
+## Extra Notes
+
+### Bids must remain valid off-chain
+
+The partially-signed bid transactions are collected by the name owner and commit
+to spending a UTXO of some value V from the blockchain. In the simplest construction,
+nothing prevents the bidder from invalidating this off-chain transaction by
+double-spending the coin they used to bid. For this reason, the name owner must
+routinely check that all submitted bids are still valid. They can try finalizing
+the highest bid and if that fails, try to finalize the second-highest bid, etc.
+More advanced constructions may be possible to more effectively prevent this
+(lightning-network-style multisigs come to mind).
+
+### A semi-custodial platform could help ensure auction finality
+
+One possible scenario with this protocol is that the name owner never finalizes
+any bids. They might have decided a "reserve price" was not met or just changed
+their mind about the sale. One technique that may be used to ensure bidders that
+a winner will be chosen by a certain time is to include a trusted moderator.
+
+Auctions could be held on a platform like Shakedex or run by an agency with a
+good reputation like Namebase or Kyokan. This agent would publish a static public
+key for their platform, and name owners would have to incorporate that key along
+with a relative locktime into the HIP-0001 locking script like so:
+
+```
+OP_TYPE
+0x09 // TRANSFER
+OP_EQUAL
+OP_IF
+  OP_IF
+    <name-owner's public key>
+  OP_ELSE
+    <timespan of auction>
+    OP_CSV
+    OP_DROP
+    <platform's public key>
+  OP_ENDIF
+  OP_CHECKSIG
+OP_ELSE
+  OP_TYPE
+  0x0a // FINALIZE
+  OP_EQUAL
+OP_ENDIF
+```
```

Comment:
is an output locking script like this "templated" enough that something like Bob Wallet can actually detect it and mark a name as "in auction" ? (i know, a n00b question). If so, is that interesting/important enough to highlight?

Reply:
It would only be able to do that if it had the preimage because only the hash of the script is on chain